avocado/utils/cpu.py

```diff
@@ -188,3 +189,56 @@ def offline(cpu):
     if _get_cpu_status(cpu):
         return 1
     return 0
+
+
+def get_cpu_idle():
+    """
+    Get current cpu idle values
+    """
+    cpus_list = cpu_online_list()
+    states = range(len(glob.glob("/sys/devices/system/cpu/cpu0/cpuidle/state*")))
+    cpu_idle_state = [[0]*len(states)]*len(cpus_list)
+    for c in cpus_list:
+        for state in states:
+            state_file = "/sys/devices/system/cpu/cpu%s/cpuidle/state%s/disable" % (c, state)
+            try:
+                cpu_idle_state[c][state] = int(open(state_file).read())
+            except Exception:
+                pass  # Ignore if states are not available
+    return cpu_idle_state
+
+
+def set_cpu_idle(state="all", disable=True, old_value=None):
+    """
+    Set/Reset cpu idle states for all cpus
+
+    :param state: cpuidle state number, default: all states
+    :param disable: whether to disable/enable given cpu idle state
+    :param old_value: whether to reset to old value if given
+    """
+    cpus_list = cpu_online_list()
+    if not old_value:
+        states = []
+        if 'all' in state:
+            states = range(1, len(glob.glob("/sys/devices/system/cpu/cpu0/cpuidle/state*")))
+        else:
+            states.append(int(state))
+        if disable:
+            value = str(1)
+        else:
+            value = str(0)
+        for c in cpus_list:
+            for state in states:
+                state_file = "/sys/devices/system/cpu/cpu%s/cpuidle/state%s/disable" % (c, state)
+                try:
+                    open(state_file, "w").write(value)
+                except Exception:
```

Comment:
Again the use of naked `Exception` is not really nice. Please use IOError instead.